src/export/StructureDefinitionExporter.ts

```diff
@@ -201,6 +221,39 @@ export class StructureDefinitionExporter implements Fishable {
     }
   }
 
+  applyDeferredRules() {
+    this.deferredRules.forEach((rules, sd) => {
+      for (const rule of rules) {
+        if (typeof rule.value === 'string') {
+          const fishedValue = this.fishForFHIR(rule.value);
+          // an inline instance will fish up an InstanceDefinition, which can be used directly.
+          // other instances will fish up an Object, which needs to be turned into an InstanceDefinition.
+          // an InstanceDefinition needs a resourceType, so check for that property.
```

Comment:
I'm a bit confused about this comment. Is an inline instance represented differently than a non-inline instance? I don't think that is the case, but I feel like I am missing something.

Reply:
Ok I think I understand, the distinction is between instances defined in FSH and those defined externally? Is that correct?

Reply:
Yes, that's precisely it. An instance defined in FSH will get us an InstanceDefinition (since that's what the InstanceExporter creates), but an instance defined externally is just some JSON we read in, and never gets turned into anything more specific than an Object.